### ISSUE TITLE
Navigating the changed files sidebar with keyboard down-arrow loses focus if scrolling is required

### ISSUE THREAD

**Issue by jstanden** [desktop/desktop#6845]
## Description

If you have enough changed files in your repository to have a scrollbar in the "Changed files" viewport on the left, you can no longer navigate the entire list of changes using only the keyboard arrows.

Once you reach a point where the list requires scrolling, the focus is stolen from the changed files and refocuses on the scrollbar itself. Subsequent up/down arrow keystrokes now only scroll the viewport, and do not select changed files nor show their diffs.

You have to press <kbd>TAB</kbd>, or click on a file with the mouse, to regain focus.

This just changed in the 1.6.2 update.

## Version

* GitHub Desktop: 1.6.2

* Operating system: macOS 10.14

## Steps to Reproduce

1. Open a repository with enough changed files to require scrolling in the file list viewport on the left. Alternatively, resize the GitHub Desktop height to force scrolling with fewer files.

2. Click on the first changed file on the left.

3. Press <kbd>Down Arrow</kbd> on the keyboard enough times to reach the bottom of the viewport and require scrolling.

### Expected Behavior

The viewport should scroll without the current file/diff losing focus. Continuing to press <kbd>Down Arrow</kbd> should navigate through the remaining files.

### Actual Behavior

Focus is stolen by the scrollbar itself. The keyboard arrow keys now only scroll the viewport and no longer navigate the list of files.

## Additional Information

![github_scrolling_keyboard 2019-02-13 17_49_01](https://user-images.githubusercontent.com/63288/52756891-b2651380-2fb7-11e9-8abf-a360ca7ac0fe.gif)

**Comment by outofambit** [comment]
Hi @jstanden I'm sorry you're hitting this. Thank you for filing such a detailed report! This is clearly a regression.

Also going to mark this as `help wanted` in case someone wants to do some more diagnostics on why and how this regressed. (May be related to changes we made in https://github.com/desktop/desktop/pull/6481.) Thanks!

**Comment by Daniel-McCarthy** [comment]
@outofambit 
> (May be related to changes we made in #6481.)

This seems to be the case. I tried rolling back this set of changes to see if the issue could still be reproduced and I could not.

I have noticed that if you select a row in the list so that it is highlighted and in focus, then scroll so that it is no longer on screen, and scroll back, it will no longer be in focus despite still being the selection. I think something similar to this is occurring when using the arrow key to select a row that is off view.

Combined with `scrollTop` being set directly it seems to be manifesting in this issue. I think a solution would be to either find a way to only set the `scrollTop` when it is needed (when switching between the History and Changes tabs). Or perhaps the stored `scrollTop` value needs to be updated when the arrow key is used to select a new row that results in changing the current scroll position.

Any ideas on how to go about the prior would certainly be appreciated. I will continue looking into this in the mean time.
